tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -586,42 +577,29 @@ SparseGlobalOrderReader::merge_result_cell_slabs(uint64_t num_cells, T cmp) {
   // For easy reference.
   auto allows_dups = array_schema_.allows_dups();
 
-  // A tile min heap, contains one ResultCoords per fragment.
-  std::vector<ResultCoords> container;
+  // A tile min heap, contains one GlobalOrderResultCoords per fragment.
+  std::vector<GlobalOrderResultCoords> container;
   container.reserve(result_tiles_.size());
-  std::priority_queue<ResultCoords, std::vector<ResultCoords>, T> tile_queue(
-      cmp, std::move(container));
-
-  std::mutex tile_queue_mutex;
+  TileMinHeap<CompType> tile_queue(cmp, std::move(container));
 
   // If any fragments needs to load more tiles.
   bool need_more_tiles = false;
 
   // Tile iterators, per fragments.
-  std::vector<std::list<ResultTileWithBitmap<uint8_t>>::iterator>
-      result_tiles_it(result_tiles_.size());
+  std::vector<TileListIt> result_tiles_it(result_tiles_.size());
 
-  // Boolean per fragment that keeps track of when a tile is used.
-  std::vector<uint8_t> result_tile_used(result_tiles_.size(), true);
-
-  // For all fragments, get the first tile.
+  // For all fragments, get the first tile in the sorting queue.
   auto status = parallel_for(
       storage_manager_->compute_tp(), 0, result_tiles_.size(), [&](uint64_t f) {
         if (result_tiles_[f].size() > 0) {
           // Initialize the iterator for this fragment.
           result_tiles_it[f] = result_tiles_[f].begin();
 
-          // Get the cell index we were processing.
-          auto cell_idx = read_state_.frag_tile_idx_[f].second;
-
           // Add the tile to the queue.
-          auto&& [st, more_tiles] = add_next_tile_to_queue(
-              f,
-              cell_idx,
-              result_tiles_it,
-              result_tile_used,
-              tile_queue,
-              tile_queue_mutex);
+          GlobalOrderResultCoords rc(
+              &*(result_tiles_it[f]), read_state_.frag_idx_[f].cell_idx_);
+          auto&& [st, more_tiles] =
+              add_next_cell_to_queue(rc, result_tiles_it, tile_queue);
           RETURN_NOT_OK(st);
           need_more_tiles = *more_tiles;
```

Comment:
How can this work? If we want to check if any fragment returned `true`, shouldn't this be `need_more_tiles |= *more_tiles` ? Or be a `vector<bool>` if we care which fragment returned what?